src/NuGet.Clients/NuGet.VisualStudio.Implementation/Extensibility/VsProjectJsonMigrator.cs

```diff
@@ -0,0 +1,51 @@
+﻿using NuGet.PackageManagement.UI;
+using NuGet.PackageManagement.VisualStudio;
+using System;
+using System.ComponentModel.Composition;
+using System.Threading.Tasks;
+
+namespace NuGet.VisualStudio
+{
+    [Export(typeof(IVsProjectJsonMigrator))]
+    class VsProjectJsonMigrator : IVsProjectJsonMigrator
+    {
+        private IVsSolutionManager _solutionManager;
+        private PumpingJTF _pumpingJtf;
+
+        [ImportingConstructor]
+        public VsProjectJsonMigrator()
+        {
+            _solutionManager = ServiceLocator.GetInstance<IVsSolutionManager>();
```

Comment:
Don't use `ServiceLocator` it leads to deadlocks. Add a constructor parameter instead, MEF will resolve the dependency for you.